### PR TITLE
feat: add `InvokeMethod2` to `Parser`

### LINKED ISSUE #7749: Add `InvokeMethod2` to front-end ASTs
Add a new node:

```scala
case class InvokeMethod2(ident: Name.Ident, exp: Expr, exps: List[Expr], loc: SourceLocation) extends Expr
```

Just below the existing `InvokeMethod`.

To begin with, we should just support the syntax:

```
.name1.name2(exps..) // Note the prefix dot
```

Here `name2` is a `Name.Ident`. We should also parse `name1` as a `Name.Ident`, but wrap it in an `Expr.Var`.

That way, we can later extend the parser without making changes to the rest.

### PR DESCRIPTION
**Pull request by chanattan**
Fixes #7749

### CODE REVIEW

**Review by magnus-madsen — COMMENTED** [review]
Great to see some quick progress. Some high-level comments:

- [x] Rename `JavaApply` to `InvokeMethod2` everywhere
- [x] Move `InvokeMethod2` to appear after `Expr.Do` everywhere
- [ ] Add support to `Parser2` and `Weeder2`

File: main/src/ca/uwaterloo/flix/language/phase/Parser.scala [hunk 1]
```diff
@@ -1089,6 +1089,10 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
       SP ~ keyword("force") ~ WS ~ RecordSelect ~ SP ~> ParsedAst.Expression.Force
     }
 
+    def JavaApply: Rule1[ParsedAst.Expression] = rule {
+      SP ~ "." ~ LowerCaseVariable ~ "." ~ Names.JavaMethod ~ ArgumentList ~ SP ~> ParsedAst.Expression.JavaApply
+    }
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Parser.scala:1094]: @herluf-ba Can you help Chanattan by sketching how this should look in Parser2/Weeder2?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Parser.scala:1094]: Strictly speaking, we should not be using ArgumentList here.

File: main/src/ca/uwaterloo/flix/language/phase/Parser.scala [hunk 2]
```diff
@@ -1175,6 +1179,16 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
       SP ~ Names.QName ~ SP ~> ParsedAst.Expression.QName
     }
 
+    def LowerCaseVariable: Rule1[ParsedAst.Expression] = {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Parser.scala:1182]: Do we need this? I think we can just use `Names.LowerCaseName`.

Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/phase/Parser.scala:1182]: That returns a Name.Ident i believe, thus the fuzz here

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Parser.scala:1182]: Lets use Name.Ident then.

Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/phase/Parser.scala:1182]: but that just postpones the trouble no? then its more work later to evaluate this `Ident`

File: main/src/ca/uwaterloo/flix/language/phase/Parser.scala (hunk 1 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Parser.scala]: Ok that is fine

**Review by magnus-madsen — APPROVED** [review]

**Comment by magnus-madsen** [comment]
Merged. Thanks!

**Pull request merged**